File: CSharp/algoExpert/isValidSubsequence.cs
using System;
using System.Collections.Generic;

public class Program {
	public static bool IsValidSubsequence(List<int> array, List<int> sequence) {
		int count = 0;
		for(int i = 0; i < array.Count; i++)
		{
			if (count < sequence.Count && array[i] == sequence[count])
			{
				count += 1;
			}
		}
		return count == sequence.Count;
	}
}

File: javascript/challenges/algoExpert/isValidSubsequence/isValidSubsequence.js
'use strict'
function isValidSubsequence(array, sequence) {
	let counter = 0;
	for(let i = 0; i < array.length; i++){
		if(array[i]===sequence[counter]){
			counter += 1;
		}
	}
	return counter === sequence.length;
}

module.exports = isValidSubsequence
